src/test/java/teammates/test/cases/browsertests/StudentProfilePageUiTest.java

```diff
@@ -338,6 +338,14 @@ private StudentProfilePage getProfilePageForStudent(String studentId) {
      */
     @AfterClass
     public void classTearDown() {
+        // The courses of test student1 account is not the same as `StudentHomePageUiTest`
+        // so it has to be cleared before running that test. This is the reason why `StudentHomePageUiTest`
+        // would fail if we don't remove the data bundle in this test.
+
+        // Since the account of user being logged in is shared in this test and `StudentHomepageUiTest`,
+        // we need to explicitly remove the data bundle of tests.
+        // The test data needs to be removed for both `StudentHomePageUiTest` and `StudentProfilePageUiTest`
+        // as the tests can run in any order.
```

Comment:
Let's add this line: ``See `BaseTestCaseWithDatastoreAccess#removeAndRestoreDataBundle(DataBundle))` for more details.``